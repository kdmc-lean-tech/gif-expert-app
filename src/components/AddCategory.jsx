import React, { useState } from 'react';
import { Form, Container } from 'react-bootstrap';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = useState('');
  const handleOnChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleOnSubmit = (e) => {
    e.preventDefault();
    if (inputValue.length > 2) {
      setCategory(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  }

  return (
    <Container className='mb-4'>
      <Form onSubmit={ handleOnSubmit } id='form'>
        <Form.Group>
          <Form.Control
            type='text'
            placeholder='Search gifs.....'
            value={ inputValue }
            onChange={ handleOnChange }
            id='input-text'
          />
        </Form.Group>
      </Form>
    </Container>
  );
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory;
