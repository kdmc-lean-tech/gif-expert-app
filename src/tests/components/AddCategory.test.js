import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('AddCategory Test', () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<AddCategory setCategory={ setCategory }/>);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategory={ setCategory }/>);
  });
  test('should render AddCategory component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should set input text in ""', () => {
    const value = wrapper.find('#input-text').prop('value');
    expect(value).toEqual('');
  });
  test('should change the input text', () => {
    const input = wrapper.find('#input-text');
    const newValue = 'test';
    input.simulate('change',
    { 
      target: {
        value: newValue
      }
    });
    const value = wrapper.find('#input-text').prop('value');
    expect(value).toEqual('test');
  });
  test('should not call setCategories function', () => {
    wrapper.find('#form').simulate('submit', { preventDefault: () => {} });
    expect(setCategory).not.toHaveBeenCalled();
  });
  test('should call the setCategory function and clear the input text', () => {
    const input = wrapper.find('#input-text');
    const newValue = 'test';
    input.simulate('change',
    { 
      target: {
        value: newValue
      }
    });
    wrapper.find('#form').simulate('submit', { preventDefault: () => {} });
    expect(setCategory).toHaveBeenCalled();
    expect(setCategory).toHaveBeenCalledWith(expect.any(Function));
    const value = wrapper.find('#input-text').prop('value');
    expect(value).toEqual('');
  });
});
