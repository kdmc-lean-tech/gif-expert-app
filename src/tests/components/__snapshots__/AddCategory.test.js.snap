// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Test should render AddCategory component 1`] = `
<Container
  className="mb-4"
  fluid={false}
>
  <Form
    id="form"
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup>
      <FormControl
        id="input-text"
        onChange={[Function]}
        placeholder="Search gifs....."
        type="text"
        value=""
      />
    </FormGroup>
  </Form>
</Container>
`;
